test/stdlib/PrintFloat.swift.gyb

```diff
@@ -578,6 +584,145 @@ PrintTests.test("Printable_CDouble") {
+  expectDescription("-1.0", asFloat16(-1.0))
+  expectDescription("1.1", asFloat16(1.1))
+  expectDescription("100.1", asFloat16(100.125))
+  expectDescription("-100.1", asFloat16(-100.125))
+
+  // Standard special numbers:
+  expectDescription("inf", Float16.infinity)
+  expectDescription("-inf", -Float16.infinity)
+  expectDescription("3.14", Float16.pi)
+  expectDescription("65504.0", Float16.greatestFiniteMagnitude)
+#if !arch(arm)
+  expectDescription("6e-08", Float16.leastNonzeroMagnitude)
+#endif
+  expectDescription("6.104e-05", Float16.leastNormalMagnitude)
+
+  // Special cases for the underlying algorithms:
+  // Smallest Float16 that requires 5 digits to print accurately
+  expectDescription("0.00010014", Float16(bitPattern: 0x0690))
+
+  // NaNs require special care in testing:
+  // NaN is printed with additional detail to debugDescription, but not description
+  expectNaN("nan", Float16.nan)
+  expectNaN("nan(0xff)", Float16(nan: 255, signaling: false))
+  expectNaN("nan(0xff)", Float16(bitPattern: 0x7eff))
+  expectNaN("-nan", -Float16.nan)
+  expectNaN("-nan(0xff)", -Float16(nan: 255, signaling: false))
+  expectFailure { expectNaN("snan", Float16.signalingNaN) }
+  expectFailure { expectNaN("-snan", -Float16.signalingNaN) }
+  expectFailure { expectNaN("snan(0xff)", Float16(nan: 255, signaling: true)) }
+  expectFailure { expectNaN("-snan(0xff)", -Float16(nan: 255, signaling: true)) }
+  expectFailure { expectNaN("snan(0xff)", Float16(bitPattern: 0x7dff)) }
+
+  expectEqual("nan", Float16.signalingNaN.description)
+  expectEqual("nan", (-Float16.signalingNaN).description)
+  expectEqual("nan", Float16(nan: 255, signaling: true).description)
+  expectEqual("nan", (-Float16(nan: 255, signaling: true)).description)
+  expectEqual("nan", Float16(bitPattern: 0x7dff).description)
+
+  // Every power of 10 should print with only a single digit '1'
+#if arch(arm)
+  let lowerBound = -7 // Or -4?  Does ARM support subnormals?
+#else
+  let lowerBound = -7
+#endif
+  for power in lowerBound ... 2 {
+    let s: String
+    if power < -4 { // Exponential form
+      s = exponentialPowerOfTen(power)
+    } else if power < 0 { // Fractional decimal form
+      s = "0." + String(repeating: "0", count: -power - 1) + "1"
+    } else { // Decimal form
+      s = "1" + String(repeating: "0", count: power) + ".0"
+    }
+    let f = Float16(s)!
+    expectDescription(s, f)
+  }
+
+  // Powers of 2
+  expectDescription("6e-08", 0x1p-24 as Float16)
+  expectDescription("1e-07", 0x1p-23 as Float16)
+  expectDescription("2.4e-07", 0x1p-22 as Float16)
+  expectDescription("5e-07", 0x1p-21 as Float16)
+  expectDescription("9.5e-07", 0x1p-20 as Float16)
+  expectDescription("1.9e-06", 0x1p-19 as Float16)
+  expectDescription("3.8e-06", 0x1p-18 as Float16)
+  expectDescription("7.6e-06", 0x1p-17 as Float16)
+  expectDescription("1.526e-05", 0x1p-16  as Float16)
+  expectDescription("3.05e-05", 0x1p-15 as Float16)
```

Comment:
Similarly, items above here are testing subnormals; are they supported on all our platforms?

Reply:
Yes. We only have hardware support for Float16 arithmetic on arm64, and on arm64 we run with subnormals enabled. On all other platforms, we use Float for the arithmetic, and Float16 inputs will never produce a subnormal result in Float.